// Author: Tim Palmer
// Created On: 02/07/2024
// Last Modified: 02/07/2024
// Program: try2
// Usage: ./try2
// Description: another try-exception example
program try2;

#include ("stdlib.hhf")
static
	u:	int32;

begin try2;
	try
		try
			stdout.put ("Enter a signed integer: ");
			stdin.get (u);
			stdout.put ("You entered: ", u, nl);
		exception (ex.ConversionError)
			stdout.put ("Your input contained illegal characters" nl);
		endtry;

		stdout.put ("Input did not fail due to a value out of range" nl);
	exception (ex.ValueOutOfRange)
		stdout.put ("The value was too large" nl);
	
	endtry;
end try2;
